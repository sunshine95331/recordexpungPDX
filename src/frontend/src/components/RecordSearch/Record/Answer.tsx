import React from 'react';
import { QuestionData } from './types';
import store from '../../../redux/store';

interface Props {
  question: QuestionData;
  selectFunction: Function;
  answer: string;
}

export function Answer(props: Props) {
  const answerId = props.question.question_id + props.answer;
  const answerData = props.question.options[props.answer];
  const isChecked = props.question.selection === props.answer;

  function handleRadioChange() {
    return store.dispatch(
      props.selectFunction(
        props.question.question_id,
        props.answer,
        answerData.edit || {}
      )
    );
  }

  return (
    <div className="dib" key={'div-' + answerId}>
      <input
        type="radio"
        name={props.question.question_id}
        id={'radio-' + answerId}
        value={props.answer}
        onChange={handleRadioChange}
        checked={isChecked}
      />
      <label htmlFor={'radio-' + answerId}>{props.answer}</label>
    </div>
  );
}
