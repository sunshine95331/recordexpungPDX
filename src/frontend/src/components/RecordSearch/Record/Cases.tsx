import React from "react";
import Case from "./Case";
import { CaseData } from "./types";

interface Props {
  cases: CaseData[];
  showEditButtons: boolean;
  whenEditing: Function;
  whenDoneEditing: Function;
}

export default class Cases extends React.Component<Props> {
  render() {
    const allCases = this.props.cases.map((caseInstance, index) => {
      return (
        <li
          key={index}
          id={"scroll-spy-trigger_case_" + caseInstance.case_number}
        >
          <Case
            case={caseInstance}
            editing={false}
            isNewCase={false}
            showEditButtons={this.props.showEditButtons}
            whenEditing={() => {
              this.props.whenEditing();
            }}
            whenDoneEditing={() => {
              this.props.whenDoneEditing();
            }}
          />
        </li>
      );
    });

    return (
      <div className="mb3">
        <ul className="list">{allCases}</ul>
      </div>
    );
  }
}
