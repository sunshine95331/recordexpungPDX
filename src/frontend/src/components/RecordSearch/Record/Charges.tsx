import React from "react";
import Charge from "./Charge";
import { ChargeData } from "./types";

interface Props {
  charges: ChargeData[];
}

export default class Charges extends React.Component<Props> {
  render() {
    const charges = this.props.charges.map((charge: ChargeData, i) => {
      return (
        <li key={charge.ambiguous_charge_id}>
          <Charge charge={charge} />
        </li>
      );
    });

    return <ul>{charges}</ul>;
  }
}
