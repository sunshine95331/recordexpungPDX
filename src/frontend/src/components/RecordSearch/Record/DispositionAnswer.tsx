import React from "react";
import { QuestionData } from "./types";
import store from "../../../redux/store";

interface Props {
  question: QuestionData;
  selectFunction: Function;
  answer: string;
  questionState: any;
  setQuestionState: any;
}

export function DispositionAnswer(props: Props) {
  const answerId = props.question.question_id + props.answer;
  const answerData = props.question.options[props.answer];
  const isChecked = props.questionState.dispositionAnswer === props.answer;

  function handleOptionRequiringTextInputFollowUp() {
    return props.setQuestionState((questionState: any) => {
      return {
        ...questionState,
        dispositionAnswer: props.answer,
      };
    });
  }

  function handleOtherDisposition() {
    props.setQuestionState((questionState: any) => {
      return {
        ...questionState,
        dispositionAnswer: props.answer,
      };
    });
    return store.dispatch(
      props.selectFunction(
        props.question.question_id,
        props.answer,
        answerData.edit || {},
        "06/01/2020"
      )
    );
  }

  function onChangeHandler() {
    switch (props.answer) {
      case "Convicted":
      case "Probation Revoked":
        return handleOptionRequiringTextInputFollowUp();
      case "Dismissed":
      case "Unknown":
        return handleOtherDisposition();
    }
  }

  return (
    <div className="dib" key={"div-" + answerId}>
      <input
        type="radio"
        name={props.question.question_id}
        id={"radio-" + answerId}
        value={props.answer}
        onChange={onChangeHandler}
        checked={isChecked}
      />
      <label htmlFor={"radio-" + answerId}>{props.answer}</label>
    </div>
  );
}
