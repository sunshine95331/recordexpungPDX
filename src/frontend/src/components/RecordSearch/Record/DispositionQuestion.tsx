import React from 'react';
import {answerDisposition} from '../../../redux/search/actions';
import {connect} from 'react-redux';
import store from '../../../redux/store';
import moment from 'moment';

interface Props {
  case_number: string;
  ambiguous_charge_id: string;
  disposition: any;
}

interface State {
  status: string;
  conviction_date: string;
  probation_revoked_date: string;
  missingFields: boolean;
  invalidDate: boolean
}

export default class DispositionQuestion extends React.Component<Props, State> {

  state: State  = {
    status: "Open",
    conviction_date: "",
    probation_revoked_date: "",
    missingFields: false,
    invalidDate: false
  }
  componentDidMount() {
    this.setState({
      status: (this.props.disposition ? this.props.disposition.status : "Open") ,
      conviction_date: (this.props.disposition ? this.props.disposition.date : ""),
      }
    )
  }

  handleChange = (e: React.BaseSyntheticEvent) => {
    this.setState<any>({
      [e.target.name]: e.target.value
    });
  };

  handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    this.validateForm().then( () => {
      if (!this.state.missingFields  && !this.state.invalidDate) {
        store.dispatch(
          answerDisposition(
            this.props.case_number,
            this.props.ambiguous_charge_id,
            this.state.status,
            (this.state.status === "Convicted" || this.state.status === "revoked" ? this.state.conviction_date : "1/1/2020"), // The date for a dismissal doesn't matter, but the backend expects something so here we are.
             this.state.probation_revoked_date)
        )
      }
    })
  };

  validateForm = () => {
    return new Promise(resolve => {
      let missingFields = false;
      if (this.state.status === "Convicted") {
        missingFields = this.state.conviction_date === "";
      } else if (this.state.status === "revoked") {
        missingFields = this.state.conviction_date === "" || this.state.probation_revoked_date === "";
      }

      let invalidDate = ( this.state.status === "Convicted" || this.state.status === "revoked") &&
        (this.state.conviction_date.length !== 0 && !moment(this.state.conviction_date, 'M/D/YYYY', true).isValid()) ||
        (this.state.probation_revoked_date.length !== 0 && !moment(this.state.probation_revoked_date, 'M/D/YYYY', true).isValid())
      ;
      this.setState(
        {
          missingFields: missingFields ,
          invalidDate: invalidDate
        },
      resolve
      );
    });
  };

  render() {
    return (
      <form className="w-100 bt bw3 b--light-purple pa3 pb1" onSubmit={this.handleSubmit}>
        <fieldset className="relative mb4">
            <legend className="fw7 mb2">What is the disposition?</legend>
            {"props: " + JSON.stringify(this.props)}
            {"state: " + JSON.stringify(this.state)}
            <div className="radio">
                <div className="dib">
                    <input id="dis" name="status" type="radio" value="Dismissed" checked={this.state.status==="Dismissed"} onChange={this.handleChange} />
                    <label htmlFor="dis">Dismissed</label>
                </div>
                <div className="dib">
                    <input id="con" name="status" type="radio" value="Convicted" checked={this.state.status==="Convicted"} onChange={this.handleChange} />
                    <label htmlFor="con">Convicted</label>
                </div>
                <div className="dib">
                    <input id="rev" name="status" type="radio" value="revoked" checked={this.state.status==="revoked"} onChange={this.handleChange} />
                    <label htmlFor="rev">Probation Revoked</label>
                </div>
                <div className="dib">
                    <input id="open" name="status" type="radio" value="Open" checked={this.state && this.state.status==="Open"} onChange={this.handleChange} />
                    <label htmlFor="open">Open</label>
                </div>
            </div>
            <div className={this.state && (this.state.status === "Convicted" || this.state.status === "revoked") ? "" : "visually-hidden"}>
              <label className="db fw6 mt3 mb1" htmlFor="n">Date Convicted <span className="f6 fw4">mm/dd/yyyy</span></label>
              <input onChange={this.handleChange} className="w5 br2 b--black-20 pa3" id="n" type="text" name="conviction_date"/>
            </div>
            <div className={this.state && this.state.status === "revoked" ? "" : "visually-hidden"}>
              <label className="db fw6 mt3 mb1" htmlFor="n">Date Probation Revoked <span className="f6 fw4">mm/dd/yyyy</span></label>
              <input onChange={this.handleChange} className="w5 br2 b--black-20 pa3" id="n" type="text" name="probation_revoked_date"/>
            </div>
            <button className="db bg-blue white bg-animate hover-bg-dark-blue fw6 br2 pv3 ph4 mt3">Submit</button>
            <div role="alert">
              <p className={(this.state && this.state.missingFields ? "" : "visually-hidden " ) + "dib bg-washed-red fw6 br3 pa3 mt3"}>Please complete all fields</p>
            </div>
            <div role="alert">
              <p className={(this.state && this.state.invalidDate ? "" : "visually-hidden " ) + "dib bg-washed-red fw6 br3 pa3 mt3"}>The date format must be MM/DD/YYYY</p>
            </div>

        </fieldset>
      </form>
    )
  }
}
