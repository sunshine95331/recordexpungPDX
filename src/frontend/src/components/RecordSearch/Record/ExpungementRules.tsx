import React from "react";
import { ExpungementRulesData } from "./types";
import {
  Disclosure,
  DisclosureButton,
  DisclosurePanel,
} from "@reach/disclosure";
interface Props {
  expungement_rules: ExpungementRulesData;
}

export default class ExpungementRules extends React.Component<
  Props,
  { open: boolean }
> {
  constructor(props: any) {
    super(props);
    this.state = {
      open: false,
    };
  }
  render() {
    const toggleOpen = () => {
      this.setState({ open: !this.state.open });
    };
    const rules = this.props.expungement_rules;
    return (
      <div className="relative mb3 connect connect-type">
        <div className="ml3 pl1">
          <Disclosure open={this.state.open} onChange={() => toggleOpen()}>
            <DisclosureButton>
              <span className="fw7">More info</span>
              {this.state.open ? (
                <span aria-hidden="true" className="fas fa-angle-up"></span>
              ) : (
                <span aria-hidden="true" className="fas fa-angle-down"></span>
              )}
            </DisclosureButton>
            <DisclosurePanel>{rules + " "}</DisclosurePanel>
          </Disclosure>
        </div>
      </div>
    );
  }
}
