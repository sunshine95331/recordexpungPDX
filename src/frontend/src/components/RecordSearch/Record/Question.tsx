import {Answer} from "./Answer";
import React from "react";
import {QuestionData} from "./types";

interface Props {
  question: QuestionData,
  selectFunction: Function,
}

export function Question(props: Props) {
  const render = () => {
    return (
      <fieldset className="relative mb4">
        <legend className="fw7 mb2">{props.question.text}</legend>
        <div className="radio">
        {Object.keys(props.question.options).map((answer: string) => {
            const id = props.question.question_id + answer;
            return (
              <Answer key={id} question={props.question} selectFunction={props.selectFunction} answer={answer} />
            );
          }
        )}
        </div>
      </fieldset>
    )
  };
  return render();
}