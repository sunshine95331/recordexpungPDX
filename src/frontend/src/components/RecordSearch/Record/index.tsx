import React, { useState } from "react";
import { useAppSelector, useAppDispatch } from "../../../redux/hooks";
import {
  startEditing,
  doneEditing,
  selectIsEditing,
} from "../../../redux/editingSlice";
import Cases from "./Cases";
import Case from "./Case";
import AddButton from "./AddButton";
import Link from "../../common/Link";

export default function Record() {
  const dispatch = useAppDispatch();
  const record = useAppSelector((state) => state.search.record);
  const isEditing = useAppSelector(selectIsEditing);
  const [enableEditing, setEnableEditing] = useState(false);
  const [addingNewCase, setAddingNewCase] = useState(false);
  const [nextNewCaseNum, setNextNewCaseNum] = useState(1);

  const createNextBlankCase = () => {
    return {
      balance_due: 0,
      birth_year: 0,
      case_detail_link: "",
      case_number: "CASE-" + ("000" + nextNewCaseNum).slice(-4),
      charges: [],
      citation_number: "",
      current_status: "",
      date: "",
      location: "",
      name: "",
      violation_type: "",
      district_attorney_number: "",
      edit_status: "ADD",
    };
  };

  const handleAddCaseClick = () => {
    dispatch(startEditing());
    setAddingNewCase(true);
  };

  return (
    <>
      {addingNewCase && (
        <div className="bg-gray-blue-2 shadow br3 overflow-auto mb3">
          <Case
            whenEditing={() => {
              dispatch(startEditing());
            }}
            whenDoneEditing={() => {
              dispatch(doneEditing());
              setAddingNewCase(false);
              setNextNewCaseNum(nextNewCaseNum + 1);
            }}
            case={createNextBlankCase()}
            editing={true}
            isNewCase={true}
            showEditButtons={!isEditing && enableEditing}
            customElementId="new-case"
          />
        </div>
      )}
      <div className="tr">
        {enableEditing ? (
          !isEditing && (
            <>
              <AddButton
                onClick={handleAddCaseClick}
                actionName={"Add"}
                text={"Case"}
              />
              <div className="pb3" />
            </>
          )
        ) : (
          <>
            <button
              className="inline-flex bg-white f6 fw5 br2 ba b--black-10 mid-gray link hover-blue pv1 ph2 mb3"
              onClick={() => {
                setEnableEditing(true);
              }}
            >
              Enable Editing
            </button>{" "}
            <Link
              to="/manual#editing"
              className=" gray link hover-blue underline"
              iconClassName="fas fa-question-circle link hover-dark-blue gray"
              hiddenText="Editing Help"
            />
          </>
        )}
      </div>

      {record?.cases && (
        <Cases
          cases={record.cases}
          showEditButtons={!isEditing && enableEditing}
          whenEditing={() => {
            dispatch(startEditing());
          }}
          whenDoneEditing={() => {
            dispatch(doneEditing());
          }}
        />
      )}
    </>
  );
}
