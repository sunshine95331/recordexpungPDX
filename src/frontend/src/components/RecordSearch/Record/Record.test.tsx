import React from "react";
import "@testing-library/jest-dom";
import { screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { appRender } from "../../../test/testHelpers";
import errorResponse from "../../../test/data/errorResponse";
import Record from ".";

const mockRequest = jest.fn((req) => new Promise(() => {}));

jest.mock("axios", () => {
  return {
    request: (request: any) => {
      mockRequest(request);
      return new Promise(() => {});
    },
  };
});

const expectedCreateCaseRequest = {
  data: {
    aliases: [
      { birth_date: "", first_name: "", last_name: "", middle_name: "" },
    ],
    demo: false,
    edits: {
      "CASE-0001": {
        summary: {
          balance_due: "0.01",
          birth_year: "1999",
          case_number: "CASE-0001",
          current_status: "Open",
          edit_status: "ADD",
          location: "Benton",
        },
      },
    },
    questions: undefined,
    today: "",
  },
  method: "post",
  url: "/api/search",
  withCredentials: true,
};

// snapshot tests for the initial state are accounted
// for in the RecordSearch tests

it("correctly renders after clicking add case button", async () => {
  const user = userEvent.setup();
  const { asFragment } = appRender(<Record />);

  await user.click(screen.getByRole("button", { name: /enable editing/i }));

  // add a case
  await user.click(screen.getByRole("button", { name: /case/i }));

  // assert the form
  expect(asFragment()).toMatchSnapshot();

  await user.click(screen.getByLabelText(/open/i));
  await user.selectOptions(
    screen.getByRole("combobox"),
    screen.getByRole("option", { name: /benton/i })
  );
  await user.click(screen.getByLabelText(/balance/i));
  // Balance already has a value of 0.00, so need to backspace first.
  await user.keyboard("{Backspace}1");
  await user.click(screen.getByLabelText(/birth year/i));
  await user.keyboard("1999");

  await user.click(screen.getByRole("button", { name: /create case/i }));

  // assert the form is closed and the Add Case button is back
  expect(asFragment()).toMatchSnapshot();

  await waitFor(() => {
    expect(mockRequest).toHaveBeenCalledWith(expectedCreateCaseRequest);
  });
});

it("displays request errors", () => {
  const { asFragment } = appRender(
    <Record record={errorResponse.record} />,
    "error"
  );
  expect(asFragment()).toMatchSnapshot();
});
