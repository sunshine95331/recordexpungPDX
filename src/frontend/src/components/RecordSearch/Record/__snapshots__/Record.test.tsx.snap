// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders after clicking add case button 1`] = `
<DocumentFragment>
  <section>
    <div
      class="bg-gray-blue-2 shadow br3 overflow-auto mb3"
    >
      <div
        class="f6 f5-l bg-gray-blue-3 shadow br3 pa1 mb4"
        id="new-case"
      >
        <div
          class="cf relative pv2 pr5 pr6-ns"
        />
        <div
          data-reach-disclosure-panel=""
          data-state="open"
          id="case-edit-CASE-0001"
          tabindex="-1"
        >
          <form
            class="pa3"
          >
             
            <fieldset
              class="mw6 pa0"
            >
              <legend
                class="visually-hidden"
              >
                Edit Case
              </legend>
              <fieldset
                class="mb3 pa0"
              >
                <legend
                  class="fw6"
                >
                  Current Status
                </legend>
                <div
                  class="radio"
                >
                  <div
                    class="dib"
                  >
                    <input
                      id="case_edit_status_open_CASE-0001"
                      name="current_status"
                      type="radio"
                      value="Open"
                    />
                    <label
                      for="case_edit_status_open_CASE-0001"
                    >
                      Open
                    </label>
                  </div>
                  <div
                    class="dib"
                  >
                    <input
                      id="case_edit_status_closed_CASE-0001"
                      name="current_status"
                      type="radio"
                      value="Closed"
                    />
                    <label
                      for="case_edit_status_closed_CASE-0001"
                    >
                      Closed
                    </label>
                  </div>
                </div>
              </fieldset>
              <div
                class="mw5 mb3"
              >
                <label
                  class="db mb1 fw6"
                  for="case_edit_location_CASE-0001"
                >
                  County
                </label>
                <div
                  class="relative mb3"
                >
                  <select
                    class="w-100 pa3 br2 bw1 b--black-20 input-reset bg-white"
                    id="case_edit_location_CASE-0001"
                    name="location"
                  >
                    <option
                      value=""
                    >
                      ---
                    </option>
                    <option
                      value="Baker"
                    >
                      Baker
                    </option>
                    <option
                      value="Benton"
                    >
                      Benton
                    </option>
                    <option
                      value="Clackamas"
                    >
                      Clackamas
                    </option>
                    <option
                      value="Clatsop"
                    >
                      Clatsop
                    </option>
                    <option
                      value="Columbia"
                    >
                      Columbia
                    </option>
                    <option
                      value="Coos"
                    >
                      Coos
                    </option>
                    <option
                      value="Crook"
                    >
                      Crook
                    </option>
                    <option
                      value="Curry"
                    >
                      Curry
                    </option>
                    <option
                      value="Deschutes"
                    >
                      Deschutes
                    </option>
                    <option
                      value="Douglas"
                    >
                      Douglas
                    </option>
                    <option
                      value="Gilliam"
                    >
                      Gilliam
                    </option>
                    <option
                      value="Grant"
                    >
                      Grant
                    </option>
                    <option
                      value="Harney"
                    >
                      Harney
                    </option>
                    <option
                      value="Hood River"
                    >
                      Hood River
                    </option>
                    <option
                      value="Jackson"
                    >
                      Jackson
                    </option>
                    <option
                      value="Jefferson"
                    >
                      Jefferson
                    </option>
                    <option
                      value="Josephine"
                    >
                      Josephine
                    </option>
                    <option
                      value="Klamath"
                    >
                      Klamath
                    </option>
                    <option
                      value="Lake"
                    >
                      Lake
                    </option>
                    <option
                      value="Lane"
                    >
                      Lane
                    </option>
                    <option
                      value="Lincoln"
                    >
                      Lincoln
                    </option>
                    <option
                      value="Linn"
                    >
                      Linn
                    </option>
                    <option
                      value="Malheur"
                    >
                      Malheur
                    </option>
                    <option
                      value="Marion"
                    >
                      Marion
                    </option>
                    <option
                      value="Morrow"
                    >
                      Morrow
                    </option>
                    <option
                      value="Multnomah"
                    >
                      Multnomah
                    </option>
                    <option
                      value="Polk"
                    >
                      Polk
                    </option>
                    <option
                      value="Sherman"
                    >
                      Sherman
                    </option>
                    <option
                      value="Tillamook"
                    >
                      Tillamook
                    </option>
                    <option
                      value="Umatilla"
                    >
                      Umatilla
                    </option>
                    <option
                      value="Union"
                    >
                      Union
                    </option>
                    <option
                      value="Wallowa"
                    >
                      Wallowa
                    </option>
                    <option
                      value="Wasco"
                    >
                      Wasco
                    </option>
                    <option
                      value="Washington"
                    >
                      Washington
                    </option>
                    <option
                      value="Wheeler"
                    >
                      Wheeler
                    </option>
                    <option
                      value="Yamhill"
                    >
                      Yamhill
                    </option>
                  </select>
                  <div
                    class="absolute pa3 right-0 top-0 bottom-0 pointer-events-none"
                  >
                    <i
                      aria-hidden="true"
                      class="fas fa-angle-down"
                    />
                  </div>
                </div>
              </div>
              <div
                class="mw5 mb3"
              >
                <label
                  class="db fw6 mb1"
                  for="case_edit_balance_CASE-0001"
                >
                  Balance
                </label>
                <div
                  class="relative"
                >
                  <div
                    class="absolute top-0 bottom-0 pl3 flex items-center"
                  >
                    <span>
                      $
                    </span>
                  </div>
                  <input
                    aria-invalid="false"
                    class="w-100 br2 b--black-20 pa3 pl4"
                    id="case_edit_balance_CASE-0001"
                    name="balance"
                    required=""
                    type="text"
                    value="0.00"
                  />
                </div>
              </div>
              <div
                class="mw5 mb4"
              >
                <label
                  class="db fw6 mb1"
                  for="case_edit_birthyear_CASE-0001"
                >
                  Birth Year 
                  <span
                    class="normal"
                  >
                    yyyy
                  </span>
                </label>
                <input
                  aria-invalid="false"
                  class="w-100 br2 b--black-20 pa3"
                  id="case_edit_birthyear_CASE-0001"
                  name="birth_year"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <div
                class="flex items-center mb3"
              >
                <button
                  class="fw6 br2 bg-blue white bg-animate hover-bg-dark-blue pa3 mr4"
                >
                  Create Case
                </button>
                <button
                  class="fw6 blue link hover-dark-blue mr4"
                >
                  Cancel
                </button>
              </div>
              <div
                class="black-70 w-100"
                role="alert"
              />
               
            </fieldset>
          </form>
        </div>
        <ul
          class="list"
        />
      </div>
    </div>
    <div
      class="tr"
    />
  </section>
</DocumentFragment>
`;

exports[`correctly renders after clicking add case button 2`] = `
<DocumentFragment>
  <section>
    <div
      class="tr"
    >
      <button
        class="tr mid-gray link hover-blue fw6"
      >
        <span
          aria-hidden="true"
          class="fas fa-plus-circle"
        />
        <span
          class="visually-hidden"
        >
          Add
        </span>
         Case
      </button>
      <div
        class="pb3"
      />
    </div>
  </section>
</DocumentFragment>
`;

exports[`displays request errors 1`] = `
<DocumentFragment>
  <p
    class="bg-washed-red mv3 pa3 br3 fw6"
    id="record_error_0"
    role="status"
  >
    first error
  </p>
  <p
    class="bg-washed-red mv3 pa3 br3 fw6"
    id="record_error_1"
    role="status"
  >
    second errorz
  </p>
  <section>
    <div
      class="bg-white shadow br3 mb3 ph3 pb3"
    >
      <div
        class="flex flex-wrap justify-end mb1"
      >
        <div
          class="flex flex-wrap items-center mv1 mr-auto"
        >
          <h2
            class="f5 fw7 mr3"
          >
            Search Summary
          </h2>
          <fieldset
            aria-label="Summary overview sort options"
            class="flex flex-wrap radio radio-sm ml1"
            dir="ltr"
            role="radiogroup"
          >
            <p>
              <input
                aria-checked="true"
                checked=""
                id="summary-overview-sort-options-charges"
                name="summary overview sort options"
                type="radio"
                value="Charges"
              />
              <label
                for="summary-overview-sort-options-charges"
              >
                Charges
              </label>
            </p>
            <p>
              <input
                aria-checked="false"
                id="summary-overview-sort-options-cases"
                name="summary overview sort options"
                type="radio"
                value="Cases"
              />
              <label
                for="summary-overview-sort-options-cases"
              >
                Cases
              </label>
            </p>
          </fieldset>
        </div>
        <button
          class="ma2 nowrap mid-gray link hover-blue fw6 br3 pv1 ph2"
        >
          <i
            aria-hidden="true"
            class="fas fa-bolt pr2"
          />
          Generate Paperwork
        </button>
        <button
          class="ma2 nowrap mid-gray link hover-blue fw6 br3 pv1 ph2"
        >
          <i
            aria-hidden="true"
            class="fas fa-download pr2"
          />
          Summary PDF
        </button>
      </div>
      <div
        class="flex-ns flex-wrap"
      >
        <div
          class="w-100 w-two-thirds-l"
        >
          <h3
            class="bt b--light-gray pt2 mb3"
          >
            <span
              class="fw7"
            >
              Cases
            </span>
             (0)
          </h3>
          <h3
            class="bt b--light-gray pt2 mb3"
          >
            <span
              class="fw7"
            >
              Charges
            </span>
             (0)
            <p
              class="f6 fw5 pt1"
            >
              Excludes traffic violations
            </p>
          </h3>
          <div
            class="mb3"
          >
            <div
              class="green bb b--light-gray lh-copy pb1"
            >
              <span
                class="fw7 mb2"
              >
                Eligible Now
              </span>
               
            </div>
            <ul
              class="list"
            />
          </div>
        </div>
        <div
          class="w-100 w-33-l ph3-l mb3"
        >
          <div
            class="mb4"
          >
            <h3
              class="fw7 bt b--light-gray pt2 mb3"
            >
              Balance due by county 
              <a
                class=" gray link hover-blue underline"
                href="/manual#paybalances"
              >
                <i
                  aria-hidden="true"
                  class="fas fa-question-circle link hover-dark-blue"
                />
                <span
                  class="visually-hidden"
                >
                  Learn more about balances
                </span>
              </a>
            </h3>
            <ul
              class="mw5 list"
            >
              None
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tr"
    >
      <button
        class="inline-flex bg-white f6 fw5 br2 ba b--black-10 mid-gray link hover-blue pv1 ph2 mb3"
      >
        Enable Editing
      </button>
       
      <a
        class=" gray link hover-blue underline"
        href="/manual#editing"
      >
        <i
          aria-hidden="true"
          class="fas fa-question-circle link hover-dark-blue gray"
        />
        <span
          class="visually-hidden"
        >
          Editing Help
        </span>
      </a>
    </div>
    <div
      class="mb3"
    >
      <ul
        class="list"
      />
    </div>
  </section>
</DocumentFragment>
`;
